pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -752,20 +746,13 @@ Optional<MessageId> ConsumerImpl::clearReceiveQueue() {
     }
 }
 
-void ConsumerImpl::increaseAvailablePermits(const ClientConnectionPtr& currentCnx, int numberOfPermits) {
-    int additionalPermits = 0;
+void ConsumerImpl::increaseAvailablePermits(const ClientConnectionPtr& currentCnx, int delta) {
+    int newAvailablePermits = addAndGet(availablePermits_, delta);
```

Comment:
This could also be done directly with: 
```suggestion
    int newAvailablePermits = availablePermits_.fetch_add(delta, std::memory_order_relaxed) + delta;
```

Reply:
I changed the code here. But I didn't specify the relaxed order. Could you explain if the relax order is enough here? If yes, how should I specify the memory order in followed `compare_exchange_weak`?

Reply:
@BewareMyPower Relaxed ordering is ok for incrementing counters (or adding a delta) since the operation is atomic and we're not relying on the ordering of multiple load/store operations. https://en.cppreference.com/w/cpp/atomic/memory_order#Relaxed_ordering